lib/field_test/participant.rb

```diff
@@ -15,7 +15,11 @@ def initialize(object)
     end
 
     def participant
-      [type, id].compact.join(":")
+      if FieldTest.legacy_participants
+        [type, id].compact.join(":")
+      else
+        type.constantize.find(id) if type.present? && id.present?
```

Comment:
If `FieldTest::legacy_participants` is `false`, then this method will never be called. Still, leaving this line just in case, but this method should probably be removed altogether when legacy participants are removed, since we'll be assigning to membership via `type` and `id`.